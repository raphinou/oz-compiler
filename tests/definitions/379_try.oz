% try-finally without catch
try
   {Show hello}
finally
   {Show final}
end

try
   try
      raise error end
   finally
      {Show final}
   end
catch error then
   {Show 'caught error exception'}
[] E then
   raise E end
end

local
   V
in
   V=try
      5
   finally
      {Show final}
   end
   {Show V}
end


try
   local
      V
   in
      V=try raise error end finally {Show final} end
      {Show V}
   end
catch error then
   {Show 'error exception caught'}
[] E then
   raise E end
end
