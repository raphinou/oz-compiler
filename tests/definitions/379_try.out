hello
final
final
'caught error exception'
final
5
final
'error exception caught'
